src/Fmt_ast.ml

```diff
@@ -68,8 +68,60 @@ let update_config c ({txt; loc}, payload) =
       !Location.warning_printer loc Caml.Format.err_formatter w ;
       c
 
-let update_config c l =
-  {c with conf= List.fold ~init:c.conf l ~f:update_config}
+let update_config c l = List.fold ~init:c l ~f:update_config
+
+let drop_while ~f s =
+  let i = ref 0 in
+  while !i < String.length s && f !i s.[!i] do
+    Int.incr i
+  done ;
+  String.sub s ~pos:!i ~len:(String.length s - !i)
+
+let maybe_disabled_k c (loc : Location.t) l f k =
+  if c.conf.disable then (
+    let loc = Source.extend_loc_to_include_attributes c.source loc l in
+    Cmts.drop_inside c.cmts loc ;
+    let s = Source.string_at c.source loc in
+    let indent_of_first_line = Position.column loc.loc_start in
+    let l = String.split ~on:'\n' s in
+    let l =
+      List.mapi l ~f:(fun i s ->
+          if i = 0 then s
+          else
+            drop_while s ~f:(fun i c ->
+                Char.is_whitespace c && i < indent_of_first_line ) )
+    in
+    k (list l "@\n" (fun s -> str s)) )
+  else f c
+
+let maybe_disabled c loc l f = maybe_disabled_k c loc l f Fn.id
+
+type block =
+  { opn: Fmt.t
+  ; pro: Fmt.t option
+  ; psp: Fmt.t
+  ; bdy: Fmt.t
+  ; cls: Fmt.t
+  ; esp: Fmt.t
+  ; epi: Fmt.t option }
+
+let empty =
+  { opn= Fn.const ()
+  ; pro= None
+  ; psp= Fn.const ()
+  ; bdy= Fn.const ()
+  ; cls= Fn.const ()
+  ; esp= Fn.const ()
+  ; epi= None }
```

Comment:
`block` and `empty` are a bit of an interruption here, maybe better to move them to the top after `type c`